cluster-agent/controllers/managed-gitops/eventloop/operation_event_loop_test.go

```diff
@@ -0,0 +1,356 @@
+			defer dbQueries.CloseDatabase()
+			defer testTeardown()
+
+			err = db.SetupForTestingDBGinkgo()
+			Expect(err).To(BeNil())
+
+			_, _, _, gitopsEngineInstance, _, err := db.CreateSampleData(dbQueries)
+			Expect(err).To(BeNil())
+
+			operationDB := &db.Operation{
+				Operation_id:            "test-operation",
+				Instance_id:             gitopsEngineInstance.Gitopsengineinstance_id,
+				Resource_id:             "test-fake-resource-id",
+				Resource_type:           db.OperationResourceType_Application,
+				State:                   db.OperationState_Waiting,
+				Operation_owner_user_id: testClusterUser.Clusteruser_id,
+			}
+
+			err = dbQueries.CreateOperation(ctx, operationDB, operationDB.Operation_owner_user_id)
+			Expect(err).To(BeNil())
+
+			By("Operation row(test-wrong-operation) doesn't exists")
+			operationCR := &operation.Operation{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name,
+					Namespace: namespace,
+				},
+				Spec: operation.OperationSpec{
+					OperationID: "test-wrong-operation",
+				},
+			}
+
+			err = task.event.client.Create(ctx, operationCR)
+			Expect(err).To(BeNil())
+
+			res, err := task.PerformTask(ctx)
+			Expect(err).To(BeNil())
+			Expect(res).To(BeFalse())
+
+		})
+
+		It("Ensure that if the kube-system namespace does not having a matching namespace uid, an error is returned", func() {
+			By("Close database connection")
+			defer dbQueries.CloseDatabase()
+			defer testTeardown()
+
+			err = db.SetupForTestingDBGinkgo()
+			Expect(err).To(BeNil())
+
+			By("'kube-system' namespace has a UID that is not found in a corresponding row in GitOpsEngineCluster database")
+			_, _, _, gitopsEngineInstance, _, err := db.CreateSampleData(dbQueries)
+			Expect(err).To(BeNil())
+			Expect(kubesystemNamespace.UID).ToNot(Equal(gitopsEngineInstance.Namespace_uid))
+
+			By("Operation row exits in database")
+			operationDB := &db.Operation{
+				Operation_id:            "test-operation",
+				Instance_id:             gitopsEngineInstance.Gitopsengineinstance_id,
+				Resource_id:             "test-fake-resource-id",
+				Resource_type:           db.OperationResourceType_Application,
+				State:                   db.OperationState_Waiting,
+				Operation_owner_user_id: testClusterUser.Clusteruser_id,
+			}
+
+			err = dbQueries.CreateOperation(ctx, operationDB, operationDB.Operation_owner_user_id)
+			Expect(err).To(BeNil())
+
+			By("Operation CR exists")
+			operationCR := &operation.Operation{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name,
+					Namespace: namespace,
+				},
+				Spec: operation.OperationSpec{
+					OperationID: operationDB.Operation_id,
+				},
+			}
+
+			err = task.event.client.Create(ctx, operationCR)
+			Expect(err).To(BeNil())
+
+			res, err := task.PerformTask(ctx)
+			Expect(err).To(BeNil())
+			Expect(res).To(BeTrue())
+
+		})
+
+		It("Ensure that if the GitopsEngineInstance's namespace_name field doesn't exist, an error is returned", func() {
```

Comment:
Same here, according story criteria it is supposed to return error, but all functions are returning nil instead of error.

Reply:
Same goes in here as above , I have written test case based on story description @jparsai